pkg/k8s/watchers/cilium_node.go

```diff
@@ -141,3 +145,37 @@ func (k *K8sWatcher) ciliumNodeInit(ciliumNPClient *k8s.K8sCiliumClient, asyncCo
 		log.Info("Disconnected from key-value store, restarting CiliumNode watcher")
 	}
 }
+
+// GetCiliumNode returns the CiliumNode "nodeName" from the local store. If the
+// local store is not initialized then it will fallback retrieving the node
+// from kube-apiserver.
+func (k *K8sWatcher) GetCiliumNode(ctx context.Context, nodeName string) (*cilium_v2.CiliumNode, error) {
+	var (
+		err                      error
+		nodeInterface            interface{}
+		exists, getFromAPIServer bool
+	)
+	k.ciliumNodeStoreMU.RLock()
+	if k.ciliumNodeStore == nil {
```

Comment:
I thought that the Agent Node watcher was always created (https://github.com/cilium/cilium/blob/master/pkg/k8s/watchers/watcher.go#L442). Why do we need this case?

Reply:
`GetCiliumNode` is called from `WaitForNodeInformation`, which happens before `ciliumNodeStore` is assigned. @aanm perhaps a comment here would be useful?